aiohttp/web.py

```diff
@@ -259,6 +266,14 @@ def startup(self):
         yield from self.on_startup.send(self)
 
     @asyncio.coroutine
+    def pre_serve(self):
+        """Causes on_pre_serve signal
+
+        Should be called just before run_forever().
+        """
+        yield from self.on_pre_serve.send(self)
```

Comment:
Im missing all of the proper test coverage of this new signal

Reply:
Your point is good. But as I mentioned in the description of this PR, I have not done all of the normal "due diligence" yet. And I won't do it until I can get some assurance that the basic design is acceptable to the maintainers.

Reply:
As a member and reviewer, I can see the point of publishing the servers as a new attribute of the app just as a simple property (only getter). Please cover it with the proper unit tests.